ministryofjustice/correspondence_tool_staff
# CT-2916 Add requester_reference

## Description

Adds requester_reference, and request_dated fields to DB.

## Self-review checklist


### Screenshots
**Before:**
<img width="1035" alt="Screenshot 2020-07-13 at 16 22 40" src="https://user-images.githubusercontent.com/13377553/87322281-1ed5a080-c525-11ea-99f4-884a4a9a8549.png">

**After:**
<img width="1012" alt="Screenshot 2020-07-13 at 16 20 55" src="https://user-images.githubusercontent.com/13377553/87322044-da4a0500-c524-11ea-81d4-b8536099b5c9.png">

case_details
<img width="1070" alt="Screenshot 2020-07-13 at 16 03 05" src="https://user-images.githubusercontent.com/13377553/87322096-e9c94e00-c524-11ea-9d29-29936f1458ad.png">


### Related JIRA tickets
[CT-2916](https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2916)

### Manual testing instructions
1) Login as Brian rix
2) fill in details
3) when at "What is the date on the request?" question fill in details - try a date in the future this should error.
4) click through to case_details references should be visible as above
5) click 'change' edit the details for the new fields
6) check that updates are reflected in case_details
